waspc/src/Generator/WebAppGenerator/RouterGenerator.hs

```diff
@@ -55,9 +66,35 @@ createRouterTemplateData wasp = RouterTemplateData
     , _pagesToImport = pages
     }
     where
-        routes = Wasp.getRoutes wasp
+        routes = map (createRouteTemplateData wasp) $ Wasp.getRoutes wasp
         pages = map createPageTemplateData $ Wasp.getPages wasp
 
+createRouteTemplateData :: Wasp -> Wasp.Route.Route -> RouteTemplateData
+createRouteTemplateData wasp route = RouteTemplateData
+    { _urlPath = Wasp.Route._urlPath route
+    , _targetComponent = determineRouteTargetComponent wasp route
+    }
+
+determineRouteTargetComponent :: Wasp -> Wasp.Route.Route -> String
+determineRouteTargetComponent wasp route =
+    maybe
+        targetPageName 
+        determineRouteTargetComponent'
+        (Wasp.Page._authRequired pageFromRoute)
+    where
+        targetPageName = Wasp.Route._targetPage route
+        -- NOTE(matija): if no page with the name specified in the route, head will fail.
+        pageFromRoute = head $ filter (((==) targetPageName) . Wasp.Page._name) (Wasp.getPages wasp)
```

Comment:
I had hard time connecting this with the variable above.
Maybe call them: targetPageName, and targetPage -> then it is obvious. Or, routeTargetPageName and routeTargetPage.

Reply:
makes sense, changed it to targetPage.